src/main/java/practice/StreamPractice.java

```diff
@@ -14,16 +27,38 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        Supplier<RuntimeException> runtExeptSupplier =
+                () -> new RuntimeException("Can't get min value from list " + numbers);
+        Optional<Integer> min = numbers.stream()
+                .flatMap(s -> Arrays.stream(s.split(",")))
+                .map(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
+                .min(Comparator.naturalOrder());
+        return min.orElseThrow(runtExeptSupplier);
     }
 
     /**
      * Given a List of Integer numbers,
      * return the average of all odd numbers from the list or throw NoSuchElementException.
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
-    public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+    public static Double getOddNumsAverage(List<Integer> numbers) {
```

Comment:
use one stream and not any Lists in this method
```suggestion
    public Double getOddNumsAverage(List<Integer> numbers) {
```